app/views/provider_interface/application_choices/index.html.erb

```diff
@@ -2,39 +2,84 @@
 
 <h1 class='govuk-heading-xl'>Applications</h1>
 
-<% if @application_choices.any? %>
-  <table class='govuk-table'>
-    <thead class='govuk-table__head'>
-      <tr class='govuk-table__row'>
-        <%= render ProviderInterface::SortableColumnHeadingComponent.new(sort_order: @sort_order, column_name: 'Name', current_sort_by: @sort_by, css_class: 'govuk-table__header govuk-!-width-one-quarter') %>
-        <%= render ProviderInterface::SortableColumnHeadingComponent.new(sort_order: @sort_order, column_name: 'Course', current_sort_by: @sort_by, css_class: 'govuk-table__header govuk-!-width-one-quarter') %>
-        <th scope='col' class='govuk-table__header govuk-!-width-one-quarter'>Status</th>
-        <%= render ProviderInterface::SortableColumnHeadingComponent.new(sort_order: @sort_order, column_name: 'Last updated', current_sort_by: @sort_by, css_class: 'govuk-table__header govuk-!-width-one-quarter govuk-table__header--numeric') %>
-      </tr>
-    </thead>
-
-    <tbody class='govuk-table__body'>
-      <% @application_choices.each do |application_choice| %>
-      <tr class='govuk-table__row'>
-        <td class='govuk-table__cell'>
-          <span class="govuk-caption-m govuk-!-font-size-16"><%= application_choice.application_form.support_reference %></span>
-          <%= govuk_link_to application_choice.application_form.full_name, provider_interface_application_choice_path(application_choice) %>
-        </td>
-        <td class='govuk-table__cell'>
-          <% if current_provider_user.providers.size > 1 %>
-            <span class='govuk-caption-m govuk-!-font-size-16'>
-              <%= application_choice.provider.name %>
-            </span>
-          <% end %>
-
-          <%= application_choice.course.name_and_code %></td>
-        <td class='govuk-table__cell'><%= render ProviderInterface::ApplicationStatusTagComponent.new(application_choice: application_choice) %></td>
-        <td class='govuk-table__cell govuk-table__header--numeric'><%= application_choice.updated_at.to_s(:govuk_date_short_month) %></td>
-      </tr>
+<% if FeatureFlag.active?('provider_application_filters') %>
+
+<div class='moj-filter-layout'>
+  <% if @filter_visible.eql?('true') %>
+    <%= render ProviderInterface::FilterComponent.new(path: :provider_interface_applications_path,
+                                                      available_filters: @available_filters,
+                                                      preselected_filters: @filter_selections,
+                                                      additional_params: { sort_by: @sort_by,
```

Comment:
This pattern is quite nice because it means the filter doesn't need to know about the sorting. As things are a change to the sorting params requires a change to this call to the filter, but that needn't be the case: we could just pass `params` here (or maybe a hash of eg `@sorting_and_filtering_options` built in the controller) and the `merge` would do the right thing. It would be possible to reuse this pattern in `ToggleFilterComponent` and `SortableColumnHeadingComponent`.

Reply:
Also this would improve the component interfaces, eg change the constructor of `ToggleFilterComponent` to be `(filter_visible:, additional_params:)` which is a bit more intention-revealing

Reply:
This is a neater way to look at this. I'll look to do this when I checkout my final branch for this feature.